File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        hash_map = {
            "type":0,
            "color":1,
            "name":2
        }
        
        index = hash_map[ruleKey]
        ans = 0
        
        for item in items:
            if item[index] == ruleValue:
                ans += 1
        
        return ans

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int answer = 0;
        int index = 0;
        switch (ruleKey) {
            case "type":
                index = 0;
                break;
            case "color":
                index = 1;
                break;
            case "name":
                index = 2;
                break;
        }
        for (List<String> item: items) {
            if (item.get(index).equals(ruleValue)) {
                answer++;
            }
        }
        return answer;
    }
}